### ISSUE TITLE
-Xint performance regressions caused by move to gcc 7

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#4210]
There is a loss in the bytecode interpreter -Xint (no JIT) performance when moving compilers from gcc 4 to gcc 7. For zlinux about 30% regression and xlinux about 17% regression. There hasn't yet been any regressions noticed when running with the JIT.

The problem has been tracked to the gcc LRA (register allocator). There is an option `-mno-lra` that can be used to recover the performance, at least on zlinux, although it is not recommended to use this debug option. This option and the change in performance hasn't yet been confirmed for xlinux.

GCC bugzilla for the register allocation problem: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88751

**Comment by pshipton** [comment]
@charliegracie @DanHeidinga @babsingh

**Comment by pshipton** [comment]
Since no regressions can be found running with the JIT, and switching from gcc 4 to 7 (for Adopt builds) is not a change that occurred in the 0.12 release cycle, removing this from the 0.12 milestone.

**Comment by pshipton** [comment]
This should be fixed by upgrading to gcc 7.5 (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88751), which according to https://gcc.gnu.org/gcc-7 was released on Nov 14.

**Comment by babsingh** [comment]
I ran the startup-footprint benchmark on Ubuntu x86 with different **gcc** versions. Here are the results:

With **gcc-4.8.0**,
```
Average Startup: 24611.80 msec (5.5% better)
Average Footprint: 143701.60 Kb
```
With **gcc-7.4.0**,
```
Average Startup: 26022.20 msec (baseline)
Average Footprint: 144126.40 Kb
```

With **gcc-7.5.0** (had to be compiled from source code),
```
Average Startup: 19848.00 msec (24% better)
Average Footprint: 146695.20 Kb
```

The startup `-Xint` regression reported in this issue is resolved in **gcc-7.5.0**. The fix impacts register allocation in gcc, which functions similarly across the different Linux architectures (x86, power and z-arch) where gcc is used for compilation. So, the issue should be fixed in all architectures. This issue can be closed once gcc-7.5.0 is installed on the build machines.

**Comment by pshipton** [comment]
The update to gcc 7.5 is completed.
https://github.com/eclipse/openj9/pull/8443
https://github.com/eclipse/openj9/pull/8559
https://github.com/AdoptOpenJDK/openjdk-build/issues/1508
